contrib/pyln-proto/pyln/proto/wire.py

```diff
@@ -59,18 +58,19 @@ def decryptWithAD(k, n, ad, ciphertext):
 
 class PrivateKey(object):
     def __init__(self, rawkey):
-        assert len(rawkey) == 32 and isinstance(rawkey, bytes)
+        if not isinstance(rawkey, bytes):
+            raise TypeError(f"rawkey must be bytes, {type(rawkey)} received")
+        elif len(rawkey) != 32:
+            raise ValueError(f"rawkey must be 32-byte long. {len(rawkey)} received")
+
         self.rawkey = rawkey
-        rawkey = int(hexlify(rawkey), base=16)
-        self.key = ec.derive_private_key(rawkey, ec.SECP256K1(),
-                                         default_backend())
+        self.key = coincurve.PrivateKey.from_int(int(hexlify(rawkey), base=16))
```

Comment:
Regarding the initialisation of key, this could be done as is using the `from_int` method (bytes -> hexlified bytes -> int) or using the `from_hex` method (bytes ->  hexlified bytes + utf-8 decoding):

self.key = coincurve.PrivateKey.from_hex((hexlify(rawkey).decode('utf-8')))

I was not sure if you'd rather an additional cast vs a decoding, so I left it as is, but it can be changed.

Reply:
How about:
```
self.key = coincurve.PrivateKey(rawkey)
```
?

Reply:
That'll do indeed.

PS: I actually didn't know about bytes.from_hex() and .hex(), TIL!

Reply:
Sorted in 259330f60f3cf3195d1234a46601e31a80d5ca98

Reply:
I've also changed all the other uses of `hexlify` in 89088e29f071e385634cd06cd09639eed7b39d89 and removed `binascii`.